### DISCUSSION TITLE
Incorrect distance future prefix for some time units in ukrainian locale

### DISCUSSION THREAD

**Discussion by integrer** [js-fns/date-fns#2876] [Q&A]
Localize token function [constantly uses](https://github.com/date-fns/date-fns/blob/c7ef5f019bb21a3f39f5d7462237978ad82ffee3/src/locale/uk/_lib/formatDistance/index.js#L31) future prefix "за", but it correct only for seconds, minutes, hours. In other time units (days, weeks, months, years) should be used prefix "через".

Also there is a [typo](https://github.com/date-fns/date-fns/blob/c7ef5f019bb21a3f39f5d7462237978ad82ffee3/src/locale/uk/_lib/formatDistance/index.js#L149): `дня -> дні`.

**Comment by tan75** [comment]
Hi @integrer 
thank you for raising this issue.
Could you please provide some reference to Ukrainian grammar rules?
I found this explanation on `month` here and it looks correct.
[onlinecorrector.com.ua](https://onlinecorrector.com.ua/uk/%D0%B7%D0%B0-%D0%BC%D1%96%D1%81%D1%8F%D1%86%D1%8C/)



> Also there is a [typo](https://github.com/date-fns/date-fns/blob/c7ef5f019bb21a3f39f5d7462237978ad82ffee3/src/locale/uk/_lib/formatDistance/index.js#L149): `дня -> дні`.
>
This is singular genitive so it looks correct to me

[Singulat genitive](https://uk.wikipedia.org/wiki/%D0%A0%D0%BE%D0%B4%D0%BE%D0%B2%D0%B8%D0%B9_%D0%B2%D1%96%D0%B4%D0%BC%D1%96%D0%BD%D0%BE%D0%BA_%D0%BE%D0%B4%D0%BD%D0%B8%D0%BD%D0%B8_%D1%96%D0%BC%D0%B5%D0%BD%D0%BD%D0%B8%D0%BA%D1%96%D0%B2_%D1%87%D0%BE%D0%BB%D0%BE%D0%B2%D1%96%D1%87%D0%BE%D0%B3%D0%BE_%D1%80%D0%BE%D0%B4%D1%83_%D0%B4%D1%80%D1%83%D0%B3%D0%BE%D1%97_%D0%B2%D1%96%D0%B4%D0%BC%D1%96%D0%BD%D0%B8)

hi @date-fns/i18n-uk wdyt?

**Reply by integrer** [reply]
> This is singular genitive so it looks correct to me

Singular generative uses for numbers ends for 2..4 except 12..14 ([here](https://github.com/date-fns/date-fns/blob/c7ef5f019bb21a3f39f5d7462237978ad82ffee3/src/locale/uk/_lib/formatDistance/index.js#L15)). In this case using rule [Matching numerals with nouns](https://studfile.net/preview/7494031/page:4/) point 2: multiple nominative. According to the [dictionary](https://slovnyk.ua/index.php?swrd=%D0%B4%D0%B5%D0%BD%D1%8C), it should be `дні`.

**Reply by integrer** [reply]
Future prefix may be really controversial. It may depend on context where duration will be used. Wouldn't it be better to let the user override this setting?
